R/dstrace.R

```diff
@@ -1,80 +1,197 @@
 
-#' dstrace
+#' A tracing utility for the GCAM Data System.
 #'
-#' @param object_name Name of object to trace
-#' @param all_data Data structure produced by a call to \code{\link{driver}}
+#' @param object_name Name of object to trace (can be either a data object or a code chunk)
+#' @param direction Trace direction (upstream, the default; downstream; or both), character
+#' @param graph Plot a directed graph of relationships? Logical
+#' @param gcam_data_map A tibble of metadata information; normally a built-in package dataset
 #' @param previous_tracelist Information about previous objects printed
-#' @param recurse Recurse to print information about precursor objects?
-#' @return Boolean (object_name found, or not)
+#' @param recurse Recurse to print information about precursor objects? Logical
+#' @param ... Extra arguments passed on to \code{\link{dstrace_plot}}
+#' @return A tibble with the trace information (object name and trace number)
+#' @details What other data products feed into some particular data system object?
+#' Conversely, to what other products does some object feed? These are the kinds
+#' of questions that \code{dstrace} can help answer.
 #' @importFrom assertthat assert_that
+#' @author BBL
 #' @export
-dstrace <- function(object_name, all_data, previous_tracelist = NULL, recurse = TRUE) {
+#' @examples
+#' dstrace("L100.FAO_ag_Exp_t")
+#' dstrace("L100.FAO_ag_Exp_t", downstream = TRUE)
```

Comment:
should this example now be `direction = "downstream"`

Reply:
Good point, thanks. I actually don't understand why R CMD CHECK didn't catch this and error.